src/github/github.installation.service.ts

```diff
@@ -0,0 +1,173 @@
+import { Injectable, Logger, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { AllowedOrganization } from './entities/allowed-organization.entity';
+import { Repository } from 'typeorm';
+import { OrganizationStatus } from 'src/common/constants';
+import { CreateOrgDto } from './dto/create.org.dto';
+import { Env } from 'src/env';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as jwt from 'jsonwebtoken';
+import { Octokit } from '@octokit/rest';
+
+@Injectable()
+export class GithubInstallationService {
+  private readonly logger = new Logger(GithubInstallationService.name);
+  constructor(
+    @InjectRepository(AllowedOrganization)
+    private allowedOrgRepository: Repository<AllowedOrganization>,
+  ) {}
+
+  async create(createOrgDto: CreateOrgDto): Promise<AllowedOrganization> {
+    const newOrg = this.allowedOrgRepository.create(createOrgDto);
+    return this.allowedOrgRepository.save(newOrg);
```

Comment:
PullProofAI

Consider adding validation logic to ensure that the `createOrgDto` contains necessary fields before creating a new organization. This can help prevent saving incomplete data.
```diff
+     // Add validation logic here
```



<!-- This is an auto-generated comment by PullProofAI -->